basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -2233,7 +2233,6 @@ private static Object handleGradebook(String sourcedid, HttpServletRequest reque
 		}
 
 		// Now read, set, or delete the grade...
-		pushAdvisor();
```

Comment:
@csev, I don't understand your use of an advisor and then the use of the Session set to admin user. Isn't the advisor completely unncecessary if you are admin user?

Reply:
Setting the user id like this in the session inside a request thread is always unsafe surely?

Reply:
I don't know this code well, but I assume this request is coming from an LTI tool not auth'ed. So doesn't the un-auth'ed request need a session to store the grade info?

Reply:
If you look deep in the Gradebook API code that sets a grade, it demands a session exist so that it can maintain a history for *who* changed a grade .  A better solution might be to add more methods to the Gradebook API - but I created and destroyed a session and wrapped it in an advisor.   This is a web service - not UI code - and it must carefully protect itself - and it does have lots of protection - which you are welcome to review.

Reply:
Why wrap it in an advisor?

Reply:
I would be happy to hop on a screen share to review this code.   Some of the GradeBook API calls blow up without a session and others work without a session.   The Advisor is there because as a web service, no one is logged in.

Reply:
https://appear.in/selective-numbat

Reply:
The advisor lets us do  things like read existing grades, etc. since the web service is just an anonymous thread.  The session is only for grade update history.   Stephen's separate idea that this should be an id other than "admin" is a good one for separation of concerns.   I set it up the to use admin because I did not want to build a complex conversion that insured that some new id that I could guarantee  *always* existed in all versions and deployments of Sakai and to make the code more easily to back and forward port.  This is also why I made it a property.

Reply:
"the web service is just an anonymous thread" until you set the current session user to "admin" ... so no need for advisor *and* the admin user!